analyzers/its/expected/Ember-MM/scraper.EmberCore.XML-{E567C031-1F7B-4637-9B3A-806988DE50CF}-S1643.json

```diff
@@ -69,32 +69,6 @@
 "id":  "S1643",
 "message":  "Use a StringBuilder instead.",
 "location":  {
-"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/Ember-MM/Addons/scraper.EmberCore.XML/XMLScraper/ScraperLib/ScraperFunction.vb#L407",
```

Comment:
TN

Reply:
We lost them because we added this check on CheckSimpleAssignment:
`context.SemanticModel.GetSymbolInfo(assigned).Symbol is ILocalSymbol`

But I'm wondering, are those TN? The rule, in this case, is suggesting changing the key of a dictionary to a StringBuilder instead of a string, I'm not aware of the behavior that may result when accessing through the indexer to retrieve values, nor if we introduce other bugs or potential perf regression with the suggested fix. I like this more cautious rule TBH. Let me know what you think

Reply:
It's a dictionary access. We need to assume that on each iteration another element is picked and then all is fine. It is a TN in my opinion.

Reply:
I added a couple of UTs both for C# and VB and marked them as FN.